yogstation/code/game/donator/borg_skins.dm

```diff
@@ -0,0 +1,189 @@
+		return TRUE
+
+//I'm overriding this so that donors will be able to pick their borg skin after choosing a module.
+/obj/item/robot_module/do_transform_delay()
+	var/mob/living/silicon/robot/R = loc
+	var/prev_lockcharge = R.lockcharge
+	sleep(1)
+	flick("[cyborg_base_icon]_transform", R)
+	R.notransform = TRUE
+	R.SetLockdown(TRUE)
+	R.anchored = TRUE
+	sleep(1)
+	for(var/i in 1 to 4)
+		playsound(R, pick('sound/items/drill_use.ogg', 'sound/items/jaws_cut.ogg', 'sound/items/jaws_pry.ogg', 'sound/items/welder.ogg', 'sound/items/ratchet.ogg'), 80, 1, -1)
+		sleep(7)
+	if(!prev_lockcharge)
+		R.SetLockdown(FALSE)
+	R.setDir(SOUTH)
+	R.anchored = FALSE
+	R.notransform = FALSE
+	R.update_headlamp()
+	R.notify_ai(NEW_MODULE)
+	if(R.hud_used)
+		R.hud_used.update_robot_modules_display()
+	SSblackbox.record_feedback("tally", "cyborg_modules", 1, R.module)
+	R.PickBorgSkin()
+
+/mob/living/silicon/ai/pick_icon() //Who the fuck wrote this shit????? Hello???
+	set category = "AI Commands"
+	set name = "Set AI Core Display"
+	if(incapacitated())
+		return
+	PickAiSkin()
+
+/mob/living/silicon/ai/proc/PickAiSkin(var/forced = FALSE)
+	icon = initial(icon)
+	if(!SSYogFeatures.DonorBorgHolder)
+		message_admins("[client.ckey] just tried to change their AI skin, but there is no borg skin holder datum! (Has the game not started yet?)")
+		to_chat(src, "An error occured, if the game has not started yet, please try again after it has. The admins have been notified about this")
+		return
+	if(is_donator(client) || forced)//First off, are we even meant to have this verb? or is an admin bruteforcing it onto a non donator for some reason?
+		var/datum/ai_skin/skins = list()
+		for(var/datum/ai_skin/S in SSYogFeatures.DonorBorgHolder.skins)
+			if(S.owner == client.ckey || !S.owner) //We own this skin.
+				skins += S //So add it to the temp list which we'll iterate through
+		var/datum/ai_skin/A //Defining A as a borg_skin datum so we can pick out the vars we want and reskin the unit
+		A = input(src,"You're a donator! Would you like to use a custom AI skin (If not, hit cancel and pick a normal one)", "Donator AI skin picker 9000", A) as null|anything in skins//Pick any datum from the list we just established up here ^^
+		if(!A)
+			return
+		if(A.name != "Cancel")
+			icon =  A.icon
+			icon_state = A.icon_state
+			to_chat(src, "You have successfully applied the skin: [A.name]")
+			return
+	else
+		var/icontype = input("Please, select a display!", "AI", null/*, null*/) in list("Clown", "Monochrome", "Blue", "Inverted", "Firewall", "Green", "Red", "Static", "Red October", "House", "Heartline", "Hades", "Helios", "President", "Syndicat Meow", "Alien", "Too Deep", "Triumvirate", "Triumvirate-M", "Text", "Matrix", "Dorf", "Bliss", "Not Malf", "Fuzzy", "Goon", "Database", "Glitchman", "Murica", "Nanotrasen", "Gentoo", "Angel")
+		if(icontype == "Clown") //To whomever it concerns. Please use a switch statement next time :)
```

Comment:
whats the point of complaining about it in a comment instead of fixing it yourself. youre the one PRing it.